File: CodeChef/Starters 178/Big_Difference.py
def solve(n, k):
    if n >= k + 1:
        a = 1
        b = k + 1
        lcm = a * b
        gcd = 1
        if abs(lcm - gcd) >= 2 * k:
            return a, b
    
    if 2*k <= n:
        a = 2*k
        b = 3*k
        gcd = k
        lcm = 6*k
        if abs(lcm - gcd) >= 2*k and a <= n and b <= n:
            return a, b
    
    if n >= 2*k and n >= 3*k:
        a = 2*k
        b = 3*k
        lcm = a * b
        gcd = k
        if abs(lcm - gcd) >= 2*k:
            return a, b
    
    if n > k:
        a = n
        b = n - k
        gcd = gcd_calc(a, b)
        lcm = (a * b) // gcd
        if abs(lcm - gcd) >= 2*k:
            return a, b
    
    return -1, -1

def gcd_calc(a, b):
    while b:
        a, b = b, a % b
    return a

t = int(input())
for _ in range(t):
    n, k = map(int, input().split())
    a, b = solve(n, k)
    print(f"{a} {b}")

File: CodeChef/Starters 178/Big_Difference.cpp
#include <bits/stdc++.h>
#include<numeric>
using namespace std;

#define ll long long
#define f(i, n) for (ll i = 0; i < n; i++)
#define ia(a, n) \
    ll a[n];     \
    f(i, n) cin >> a[i]
#define iv(v, n) \
    vector<ll> v(n); \
    f(i, n) cin >> v[i]
#define MOD (1000000007)
#define INF 1000000000000000000LL // Infinity for ll
#define mp make_pair
#define nline '\n'
#define yes cout << "YES\n"
#define no cout << "NO\n"

// read question properly
// don't forget newlines!!!!!!
// take care about cin >> t;
// comment the optimization before debugging
// ALWAYS USE FIXED << SETPRECISION WHILE OUTPUTTING FLOATS

void solve()
{
    ll n, k;
    cin >> n >> k;
    if (k >= n)
    {
        cout << -1 << ' ' << -1 << nline;
        return;
    }

    if (n - k <= 50)
    {
        for (ll a = 1; a + k <= n; a++)
        {
            for (ll b = a + k; b <= n; b++)
            {
                if ((a * b) / __gcd(a, b) - __gcd(a, b) >= 2 * k)
                {
                    cout << a << ' ' << b << nline;
                    return;
                }
            }
        }
        cout << -1 << ' ' << -1 << nline;
    }
    else
    {
        ll a = n;
        for (ll b = 2;; b++)
        {
            if (__gcd(a, b) == 1)
            {
                cout << a << ' ' << b << nline;
                return;
            }
        }
    }
}
int main()
{
#ifdef thepratholic
    freopen("input.txt", "r", stdin);
    freopen("output.txt", "w", stdout);
    clock_t T = clock();
#endif

    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    cout.tie(NULL);

    long long t = 1;
    cin >> t;

    while (t--)
    {
        solve();
    }

#ifdef thepratholic
    cout << "\nTime taken: " << ((float)(clock() - T)) / CLOCKS_PER_SEC << " seconds";
#endif
    return 0;
}